explain-subqueries.md

```diff
@@ -0,0 +1,146 @@
+```
+
+```sql
++-----------------------------+-----------+-----------+------------------------------+--------------------------------------------------------------------------------------------------------+
+| id                          | estRows   | task      | access object                | operator info                                                                                          |
++-----------------------------+-----------+-----------+------------------------------+--------------------------------------------------------------------------------------------------------+
+| MergeJoin_9                 | 45446.40  | root      |                              | semi join, left key:test.t1.id, right key:test.t2.t1_id, other cond:ne(test.t2.t1_id, test.t1.int_col) |
+| ├─IndexReader_24(Build)     | 180000.00 | root      |                              | index:IndexFullScan_23                                                                                 |
+| │ └─IndexFullScan_23        | 180000.00 | cop[tikv] | table:t2, index:t1_id(t1_id) | keep order:true                                                                                        |
+| └─TableReader_22(Probe)     | 56808.00  | root      |                              | data:Selection_21                                                                                      |
+|   └─Selection_21            | 56808.00  | cop[tikv] |                              | ne(test.t1.id, test.t1.int_col)                                                                        |
+|     └─TableFullScan_20      | 71010.00  | cop[tikv] | table:t1                     | keep order:true                                                                                        |
++-----------------------------+-----------+-----------+------------------------------+--------------------------------------------------------------------------------------------------------+
+6 rows in set (0.00 sec)
+```
+
+From the result above, you can see that TiDB uses a `Semi Join` algorithm. Semi-join differs from `INNER JOIN`: semi-join only permits the first value on the right key (`t2.t1_id`), which means that the duplicates are eliminated as a part of the join operator task. The join algorithm is also Merge Join, which is like an efficient zipper-merge as the operator reads data from both the left and the right side in sorted order.
+
+The original statement is considered a _correlated subquery_, since the subquery refers to a column (`t1.int_col`) that exists outside of the subquery. However, the output of `EXPLAIN` shows the execution plan after the [subquery decorrelation optimization](/correlated-subquery-optimization.md) has been applied. The condition `t1_id != t1.int_col` is rewritten to `t1.id != t1.int_col`. This is a lot more efficient, because TiDB can perform this check in `└─Selection_21` as it is reading data from the table `t1`.
```

Comment:
+ Does `the original statement` mean `SELECT t1_id FROM t2 WHERE t1_id != t1.int_col` or `SELECT * FROM t1 WHERE id IN (SELECT t1_id FROM t2 WHERE t1_id != t1.int_col);`?
+ `since` could be ambiguous in tech documentation.
+ It looks like `t1.int_col` is right inside `SELECT t1_id FROM t2 WHERE t1_id != t1.int_col);` ...
```suggestion
The original statement is considered a _correlated subquery_, because the subquery refers to a column (`t1.int_col`) that exists outside of the subquery. However, the output of `EXPLAIN` shows the execution plan after the [subquery decorrelation optimization](/correlated-subquery-optimization.md) has been applied. The condition `t1_id != t1.int_col` is rewritten to `t1.id != t1.int_col`. This rewriting makes the execution a lot more efficient, because TiDB can perform this check in `└─Selection_21` as it is reading data from the table `t1`.
```